macros/sql_gen/gen_joined_metrics_cte.sql

```diff
@@ -139,11 +139,21 @@
     {%- else %}
         join_metrics__{{previous_cte_number}}.*
     {%- endif %}
-    {%- for metric in metric_tree.ordered_expression_set %}
+    {%- for metric in metric_tree.expression_set %}
```

Comment:
there was a merge conflict here at some point -- feels like this wasn't supposed to get changed? @callum-mcdata

Reply:
Hmm that is a bit odd. In the end it shouldn't matter because the key values in `ordered_expression_set` will match the values in the list that is `expression_set` but as long as the tests pass I'm fine with your override

Reply:
I think `expression_set` actually makes more sense here rather than the key val pairs in the ordered set!